pkg/ocm/client.go

```diff
@@ -0,0 +1,93 @@
+package ocm
+
+import (
+	"fmt"
+	"time"
+
+	sdkClient "github.com/openshift-online/ocm-sdk-go"
+	"github.com/patrickmn/go-cache"
+)
+
+type Client struct {
+	config     *Config
+	logger     sdkClient.Logger
+	connection *sdkClient.Connection
+	Cache      *cache.Cache
+
+	Authentication OCMAuthentication
+	Authorization  OCMAuthorization
+}
+
+type Config struct {
+	BaseURL      string `envconfig:"OCM_BASE_URL" default:""`
+	ClientID     string `envconfig:"OCM_SERVICE_CLIENT_ID" default:""`
+	ClientSecret string `envconfig:"OCM_SERVICE_CLIENT_SECRET" default:""`
+	SelfToken    string `envconfig:"OCM_SELF_TOKEN" default:""`
+	Debug        bool   `envconfig:"OCM_DEBUG" default:"false"`
+}
+
+func NewClient(config Config) (*Client, error) {
+	// Create a logger that has the debug level enabled:
+	logger, err := sdkClient.NewGoLoggerBuilder().
+		Debug(config.Debug).
+		Build()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to build OCM logger: %s", err.Error())
+	}
+	client := &Client{
+		config: &config,
+		logger: logger,
+		Cache:  cache.New(1*time.Minute, 30*time.Minute),
+	}
+	client.Authentication = &authentication{
+		client: client,
+	}
+	client.Authorization = &authorization{
+		client: client,
+	}
+	return client, nil
+}
+
+// NewConnection cretaes a new connection
+func (c *Client) NewConnection() error {
```

Comment:
as we spoke on chat, this should return the connection and not save in inside the client instance